### DISCUSSION TITLE
Reinforcement Learning.jl in a RTS

### DISCUSSION THREAD

**Discussion by ashwanirathee** [JuliaReinforcementLearning/ReinforcementLearning.jl#292] [General]
Hey @findmyway, I want to ask if there are any existing projects that use ReinforcementLearning.jl in a real game like Minecraft, 0 a.d., dota2 etc. I am quite interested in this topic and would love to explore this.
For mincraft, there is proper malmo project: https://github.com/microsoft/malmo , project is under MIT license but mincraft itself isn't itself open source. Malmo has like some pieces of mincraft to create scenarios.
There is none for 0 a.d. but its kind of an equivalent of  AOE and it has clean nice scenario editor. And it's open source, so worth a look imo.

I wanted to ask about your thoughts on this topic 😄

**Comment by findmyway** [comment]
Hi @ashwani-rathee ,

Until now, I'm not aware of any projects that use this package to play games as you listed above.
I've investigated malmo about a year ago. TBH, I really didn't know how to write a native julia wrapper for it. (Just take a look at it again, I'm wondering if it is still been maintained?)
So generally, if we can communicate with those games in Julia, them we can always apply algorithms in this package somehow.

**Reply by ashwanirathee** [reply]
Malmo is not maintained now,but all the python examples run well. It uses Minecraft 1.11.2 is which is fairly old version but the thing somehow works still.
Python has something like MalmoPython.so that python uses to communicate with game. I will need to read how things work in it.

**Reply by ashwanirathee** [reply]
Well, Good thing is that cmu-rl maintains a fork : https://github.com/cmu-rl/malmo
https://techgaun.github.io/active-forks/index.html#https://github.com/microsoft/malmo

**Reply by findmyway** [reply]
I see. So the first step is to simply use the [`GymEnv`](https://github.com/JuliaReinforcementLearning/ReinforcementLearning.jl/blob/98f5f20a8eaf51ea0ba1c9427bd19eb22c0ec655/src/ReinforcementLearningEnvironments/src/environments/3rd_party/gym.jl#L4) to verify that we can use it.

**Reply by ashwanirathee** [reply]
I was exploring malmoenv for python today, I made a script to run MalmoEnv(which is in python)  in julia 
run.jl with instructions: https://gist.github.com/ashwani-rathee/f52cfac73b98c6100c5e212f903f1cf7
MalmoEnv is written in python, so that might take longer to translate and implement, I am able to get the data, reward and some other data

**Reply by ashwanirathee** [reply]
@findmyway 
Should I implement this malmoenv for julia: https://github.com/microsoft/malmo/tree/master/MalmoEnv/malmoenv?

**Reply by findmyway** [reply]
Could you also run some benchmarks to estimate the time cost of each step?
Note that there are some tricks to reduce the cost of calling Python code in Julia. You can read the code in [GymEnv](https://github.com/JuliaReinforcementLearning/ReinforcementLearning.jl/blob/master/src/ReinforcementLearningEnvironments/src/environments/3rd_party/gym.jl#L54) for example.

**Comment by findmyway** [comment]
You might be interested in this: https://minerl.io/
